src/model/transaction/TransferTransaction.ts

```diff
@@ -280,22 +280,13 @@ export class TransferTransaction extends Transaction {
      * @param aggregateTransactionIndex Transaction index for aggregated transaction
      * @returns {TransferTransaction}
      */
-    resolveAliases(statement: Statement, aggregateTransactionIndex: number = 0): TransferTransaction {
+    public resolveAliases(statement: Statement, aggregateTransactionIndex: number = 0): TransferTransaction {
         const transactionInfo = this.checkTransactionHeightAndIndex();
-        return new TransferTransaction(
-            this.networkType,
-            this.version,
-            this.deadline,
-            this.maxFee,
-            statement.resolveAddress(this.recipientAddress,
-                transactionInfo.height.toString(), transactionInfo.index, aggregateTransactionIndex),
-            this.mosaics.map((mosaic) =>
-                statement.resolveMosaic(mosaic, transactionInfo.height.toString(),
-                    transactionInfo.index, aggregateTransactionIndex)),
-            this.message,
-            this.signature,
-            this.signer,
-            this.transactionInfo,
-        );
+        return Object.assign({__proto__: Object.getPrototypeOf(this)}, this,
+            {
+                recipientAddress: statement.resolveAddress(this.recipientAddress,
+                    transactionInfo.height.toString(), transactionInfo.index, aggregateTransactionIndex),
+                mosaics: this.mosaics.map((mosaic) => statement.resolveMosaic(mosaic, transactionInfo.height.toString(),
+                    transactionInfo.index, aggregateTransactionIndex))});
```

Comment:
Can this be changed for :

```
return {}.extend(this, {...}) as TransferTransaction
```

-- 

I believe with extend, it respect code clarity guidelines as with your `Object.assign` it adds the knowledge of _prototype_ that we may avoid.

What you are trying to do here: 

- remove the initialisation of objects by passing each fields (`return new TransferTransaction(fields here)`)
- return an **extended** object
- return an object **type cast** as TransferTransaction

I believe for this matter, the `{}.extend() as SpecializedClass` adds more clarity and does same job ; Would need small test to make sure, just wanted this to be discussed.

Reply:
They both should do the merge. but Object.assign can return undefiled if the object value merged is undefined. but extend might pick the original (here in our case the unresolved) value if new value undefined. 
Don't think it will make much differences here, as the resolution process should throw error for undefined anyway, but IMO, returning undefined can be better as it avoids silent fail just in case

Reply:
Should you be using the object spread operator?

https://basarat.gitbooks.io/typescript/docs/spread-operator.html